trulens_eval/trulens_eval/streamlit.py

```diff
@@ -0,0 +1,255 @@
+        ): row.feedback_json.get("higher_is_better", True)
+        for _, row in feedback_defs.iterrows()
+    }
+
+    if df.empty:
+        st.write("No records yet...")
+        return
+
+    df = df.sort_values(by="app_id", inplace=True)
+
+    if df.empty:
+        st.write("No records yet...")
+
+    if app_ids is None:
+        app_ids = list(df.app_id.unique())
+
+    for app_id in app_ids:
+        app_df = df.loc[df.app_id == app_id]
+        if app_df.empty:
+            continue
+        app_str = app_df["app_json"].iloc[0]
+        app_json = json.loads(app_str)
+        metadata = app_json.get("metadata")
+        st.header(app_id, help=draw_metadata(metadata))
+        app_feedback_col_names = [
+            col_name for col_name in feedback_col_names
+            if not app_df[col_name].isna().all()
+        ]
+        col1, col2, col3, col4, *feedback_cols = st.columns(
+            5 + len(app_feedback_col_names)
+        )
+        latency_mean = (
+            app_df["latency"].
+            apply(lambda td: td if td != MIGRATION_UNKNOWN_STR else None).mean()
+        )
+
+        col1.metric("Records", len(app_df))
+        col2.metric(
+            "Average Latency (Seconds)",
+            (
+                f"{millify(round(latency_mean, 5), precision=2)}"
+                if not math.isnan(latency_mean) else "nan"
+            ),
+        )
+        col3.metric(
+            "Total Cost (USD)",
+            f"${millify(round(sum(cost for cost in app_df.total_cost if cost is not None), 5), precision = 2)}",
+        )
+        col4.metric(
+            "Total Tokens",
+            millify(
+                sum(
+                    tokens for tokens in app_df.total_tokens
+                    if tokens is not None
+                ),
+                precision=2
+            ),
+        )
+
+        for i, col_name in enumerate(app_feedback_col_names):
+            mean = app_df[col_name].mean()
+
+            st.write(
+                styles.stmetricdelta_hidearrow,
+                unsafe_allow_html=True,
+            )
+
+            higher_is_better = feedback_directions.get(col_name, True)
+
+            if "distance" in col_name:
+                feedback_cols[i].metric(
+                    label=col_name,
+                    value=f"{round(mean, 2)}",
+                    delta_color="normal"
+                )
+            else:
+                cat = CATEGORY.of_score(mean, higher_is_better=higher_is_better)
+                feedback_cols[i].metric(
+                    label=col_name,
+                    value=f"{round(mean, 2)}",
+                    delta=f"{cat.icon} {cat.adjective}",
+                    delta_color=(
+                        "normal" if cat.compare(
+                            mean, CATEGORY.PASS[cat.direction].threshold
+                        ) else "inverse"
+                    ),
+                )
+
+        st.markdown("""---""")
+
+@st.experimental_fragment(run_every=2)
```

Comment:
@sfc-gh-jreini `experimental_fragment` only came out in v1.33.0, while it looks like we have `streamlit          >= 1.32.2` - we might want to bump the version